KELP_LIMITATIONS.md

```diff
@@ -0,0 +1,31 @@
+# KELP Account Limitations & Account Abstraction Challenges on Sui
+
+## Current State
+
+KELP (Key-Loss Protection) provides a recovery mechanism for Sui accounts, but the current implementation faces significant limitations that prevent it from functioning as a true account abstraction solution.
+These limitations stem from Sui's object-centric model where the `Kelp` object acts as a container rather than an autonomous account.
+
+## Key Limitations
```

Comment:
I prefer adding real examples and implications. For instance

- what's the implications of not being able to self-sign transactions? Do we always rely on a sponsor, and if not, let's assume we had a multi-user account, what would be the gas management strategy without a 3rd party?
- No direct system function access - give the example of swap
- Extend the system function access to all contracts, for instance we could fix at protocol level functions like swap, but what do we do for contracts we don't control? what about backwards compatibility?
- Manual claiming required. The issue is complex account management + the fact that finality is now requiring two transactions.
- Restricted transfer capabilities, Complex object handling: what's the complexity on current wallets supporting it?